build-id-versioning/worker/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"github.com/pborman/uuid"
+	build_id_versioning "github.com/temporalio/samples-go/build-id-versioning"
+	"sync"
+
+	"log"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+func main() {
+	c, err := client.Dial(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+
+	taskQueue := "build-id-versioning-" + uuid.New()
+	log.Println("Using Task Queue name: ", taskQueue, "(Copy this!)")
+
+	// We will start a handful of workers, each with a different build identifier.
+	wg := sync.WaitGroup{}
+	createAndRunWorker(c, taskQueue, "1.0", build_id_versioning.SampleChangingWorkflowV1, &wg)
+	createAndRunWorker(c, taskQueue, "1.1", build_id_versioning.SampleChangingWorkflowV1b, &wg)
+	createAndRunWorker(c, taskQueue, "2.0", build_id_versioning.SampleChangingWorkflowV2, &wg)
+	wg.Wait()
+}
+
+func createAndRunWorker(c client.Client, taskQueue, buildID string, workflowFunc interface{}, wg *sync.WaitGroup) {
```

Comment:
Can take a callback to not harm the static analyzer here

```suggestion
func createAndRunWorker(c client.Client, taskQueue, buildID string, wg *sync.WaitGroup, register func(w *worker.Worker)) {
```

And then just call as:

```go
createAndRunWorker(c, taskQueue, "1.0", &wg, func (w *worker.Worker) {
    w.RegisterWorkflowWithOptions(
        build_id_versioning.SampleChangingWorkflowV1, 
        workflow.RegisterOptions{Name: "SampleChangingWorkflow"},
    )
})
```

Reply:
I merged the fix I needed anyway. Doing this takes away part of the point which is that the same name always gets used

Reply:
Why is this harming the static analyzer?

Reply:
It can't follow higher-order function refs to go analyze them, but, it's not exactly important that it does here.

Reply:
Technically we could make it do so (we would taint any expression that _becomes_ a workflow function as needing to be deterministic, and then we'd know this param would need to be), but it's not the simplest or that common.